client/widmgr/widmgr.go

```diff
@@ -61,11 +61,24 @@ type WIDMgr struct {
 func NewWIDMgr(signer IdentitySigner, a *structs.Allocation, logger hclog.Logger) *WIDMgr {
 	widspecs := map[string][]*structs.WorkloadIdentity{}
 	tg := a.Job.LookupTaskGroup(a.TaskGroup)
+
+	for _, service := range tg.Services {
+		if service.Identity != nil {
+			widspecs[service.Identity.Name] = []*structs.WorkloadIdentity{service.Identity}
```

Comment:
There's something 🐟-y going on here: if we key by `service.Identity.Name`, other parts of the code aren't able to find these. Something looks up by `task.Name`, so we need to investigate.